File: dagger/src/main/java/com/dagger/mvp/view/MVPComponent.java
package com.dagger.mvp.view;


import com.dagger.application.App;
import com.dagger.application.ApplicationComponent;
import com.dagger.mvp.model.RegisterModel;
import com.dagger.mvp.widget.MVPActivity;

import dagger.Component;

/**
 * by y on 2017/5/31.
 */
@Component(dependencies = ApplicationComponent.class, modules = RegisterModel.class)
public interface MVPComponent {
    void register(MVPActivity activity);

    App getApplication();
}

File: UtilsSample/Dagger/src/main/java/sample/util/develop/android/dagger/mvp/view/MVPComponent.kt
package sample.util.develop.android.dagger.mvp.view


import dagger.Component
import sample.util.develop.android.dagger.application.App
import sample.util.develop.android.dagger.application.ApplicationComponent
import sample.util.develop.android.dagger.mvp.model.RegisterModel
import sample.util.develop.android.dagger.mvp.widget.MVPActivity

/**
 * by y on 2017/5/31.
 */
@Component(dependencies = [ApplicationComponent::class], modules = [RegisterModel::class])
interface MVPComponent {
    val application: App
    fun register(activity: MVPActivity)
}
